### PR TITLE
[chrome] Add missing definition of chrome.action.setIcon

### PR DESCRIPTION
**Pull request by ghost**
Add missing definition of `chrome.action.setIcon` in Manifest V3.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://developer.chrome.com/docs/extensions/reference/action/#method-setIcon>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/chrome/index.d.ts
Comment (fregante) [inline types/chrome/index.d.ts:320]: In the non-promisified API below, `callback` should not be optional. When it's omitted, the function returns a Promise.

**Review by fregante — APPROVED** [review]
It can be merged after this change, but I think other maintainers also need to approve this PR.

File: types/chrome/index.d.ts
Comment (fregante) [reply types/chrome/index.d.ts:320]: See https://github.com/fregante/DefinitelyTyped/pull/1

**Review by matcarlson — CHANGES_REQUESTED** [review]

File: types/chrome/index.d.ts
Comment (LightAPIs) [reply types/chrome/index.d.ts:320]: > +1, that should be preferred. In non-promise versions, callback should not be optional. Can you update line 329 to make the callback not optional?

I've updated it.

**Review by matcarlson — APPROVED** [review]

**Comment by LightAPIs** [comment]
Ready to merge

**Review by 333ize — APPROVED** [review]

**Review by 333ize — APPROVED** [review]

**Pull request merged**
